Configuring the colorscheme makes my formatter and lsp exit

When I have:

```R
colorout::setOutputColors(
# ...
)
```
Set to anything in Rprofile my LSP ([`r-languageserver`](https://github.com/REditorSupport/languageserver)) and formatter ([`styler`](https://github.com/r-lib/styler)) just exit and refuse to work.

Here are the errors:

`r-languageserver`:

```
[ERROR][2024-02-24 19:40:25] .../vim/lsp/rpc.lua:734	"rpc"	"r-languageserver"	"stderr"	'Error in .C("colorout_SetZero", z, PACKAGE = "colorout") : \n  "colorout_SetZero" not available for .C() for package "colorout"\nCalls: <Anonymous> -> setZero\nExecution halted\n'
```

`styler`:

```
19:40:38[ERROR] Formatter 'my_styler' error: Error in .C("colorout_SetZero", z, PACKAGE = "colorout") : 
  "colorout_SetZero" not available for .C() for package "colorout"
Calls: <Anonymous> -> setZero
Execution halted
```

When I do not configure the colorscheme everything works perfectly, but obviously I wish I could configure my colorscheme and have everything working. Any ideas how to fix this?

I should probably also provide some additional context. I am running Fedora Linux 39 aarch64, and editor is neovim. I installed manually from source into a local directory.

Build logs:

```
* installing *source* package ‘colorout’ ...
** using staged installation
** libs
using C compiler: ‘gcc (GCC) 13.2.1 20231205 (Red Hat 13.2.1-6)’
gcc -I"/usr/include/R" -DNDEBUG   -I/usr/local/include    -fPIC  -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer   -c colorout.c -o colorout.o
colorout.c: In function ‘isnumber’:
colorout.c:87:50: warning: comparison of constant ‘45’ with boolean expression is always false [-Wbool-compare]
   87 |     if(i > 0 && !iswhitespace(b[i-1]) && !b[i-1] == '-')
      |                                                  ^~
colorout.c:87:50: warning: logical not is only applied to the left hand side of comparison [-Wlogical-not-parentheses]
colorout.c:87:42: note: add parentheses around left hand side expression to silence this warning
   87 |     if(i > 0 && !iswhitespace(b[i-1]) && !b[i-1] == '-')
      |                                          ^~~~~~~
      |                                          (      )
gcc -shared -L/usr/lib64/R/lib -Wl,-z,relro -Wl,--as-needed -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -Wl,--build-id=sha1 -o colorout.so colorout.o -L/usr/lib64/R/lib -lR
installing to /home/user/tmp/test/00LOCK-colorout/00new/colorout/libs
** R
** inst
** byte-compile and prepare package for lazy loading
Warning message:
.onUnload failed in unloadNamespace() for 'colorout', details:
  call: .C("colorout_noColorOutput", PACKAGE = "colorout")
  error: "colorout_noColorOutput" not available for .C() for package "colorout"
** help
*** installing help indices
  converting help for package ‘colorout’
    finding HTML links ... done
    ColorOut                                html
    addPattern                              html
    colorout-package                        html
    deletePattern                           html
    isColorOut                              html
    listPatterns                            html
    noColorOut                              html
    setOutputColors                         html
    setOutputColors256                      html
    setZero                                 html
    show256Colors                           html
    unsetZero                               html
*** copying figures
** building package indices
** testing if installed package can be loaded from temporary location
** checking absolute paths in shared objects and dynamic libraries
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (colorout)
```

I am not sure if the error and warnings in the logs are related to this issue, but I still consider that the issue may come from compilation given that this is C code and I am running aarch64 with 16k page size which is pretty uncommon.

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -1,5 +1,5 @@
 Package: colorout
-Version: 1.3-0.1
+Version: 1.3-0.2
 Date: 2023-09-26
 Title: Colorize R Output on Terminal Emulators
 Author: Jakson Aquino and Dominique-Laurent Couturier

--- a/R/colorout.R
+++ b/R/colorout.R
@@ -10,6 +10,8 @@
 
 
 .onLoad <- function(libname, pkgname) {
+    library.dynam("colorout", pkgname, libname, local = FALSE);
+
     if(is.null(getOption("colorout.anyterm")))
         options(colorout.anyterm = FALSE)
     if(is.null(getOption("colorout.dumb")))
@@ -20,13 +22,18 @@
         options(colorout.notatty = FALSE)
     if(is.null(getOption("colorout.verbose")))
         options(colorout.verbose = 0)
-    return(invisible(NULL))
+
+    msg <- testTermForColorOut()
+    if(msg != "OK" && getOption("colorout.verbose") > 0){
+        msg <- paste(gettext("The R output will not be colorized because it seems that your terminal does not support ANSI escape codes.", domain = "R-colorout"),
+                     msg)
+        warning(msg, call. = FALSE, immediate. = TRUE)
+    }
 }
 
 .onAttach <- function(libname, pkgname) {
     msg <- testTermForColorOut()
     if (msg == "OK") {
-        library.dynam("colorout", pkgname, libname, local = FALSE);
         ColorOut()
     } else if(getOption("colorout.verbose") > 0){
         msg <- paste(gettext("The R output will not be colorized because it seems that your terminal does not support ANSI escape codes.",
@@ -84,10 +91,7 @@ noColorOut <- function()
 
 isColorOut <- function()
 {
-    is_it <- try(.Call("colorout_is_enabled", PACKAGE = "colorout"), silent = TRUE)
-    if (inherits(is_it, "try-error"))
-        return(FALSE)
-    return(is_it)
+    .Call("colorout_is_enabled", PACKAGE = "colorout")
 }
 
 GetColorCode <- function(x, name)

--- a/tests/test-isColorOut.R
+++ b/tests/test-isColorOut.R
@@ -1,0 +1,19 @@
+if (requireNamespace("colorout", quietly = TRUE)) {
+    #options("colorout.verbose" = 2)
+
+    # ensure colorout is not attached
+    pkg <- "package:colorout"
+    if (pkg %in% search()) {
+        detach("package:colorout")
+    }
+
+    # isColorOut() should return FALSE if package is not attached
+    stopifnot(!colorout::isColorOut())
+
+    # isColorOut() should return TRUE if package is attached
+    if (interactive()) {
+        # colorout only works in interactive sessions
+        require("colorout", quietly = TRUE)
+        stopifnot(isColorOut())
+    }
+}

